src/streaming/controllers/AlternativeMpdController.js

```diff
@@ -180,21 +177,32 @@ function AlternativeMpdController() {
 
     function _onDashPlaybackTimeUpdated(e) {
         try {
-            if (currentEvent) {
-                if (currentEvent.type == 'dynamic') { return; }
-                if (currentEvent.duration <= e.time && Math.round(e.time - lastTimestamp) != 0) {
-                    _switchBackToMainContent(currentEvent);
-                }
-            }
-            else {
-                const currentTime = e.time;
+            const currentTime = e.time;
+            if (!currentEvent) {
                 lastTimestamp = e.time;
                 const event = _getCurrentEvent(currentTime);
 
                 if (event && !isSwitching && !currentEvent) {
                     currentEvent = event;
                     _switchToAlternativeContent(event);
                 }
+                return;
+            } 
+
+            if (currentEvent.type == 'dynamic') { 
+                return; 
+            }
+
+            if (Math.round(e.time - lastTimestamp) != 0) {
+                return
+            }
+
+            const { presentationTime, duration, clip } = currentEvent;
+
+            if (clip && lastTimestamp + e.time >= presentationTime + duration) {
+                _switchBackToMainContent(currentEvent);
+            } else if (duration <= e.time) {
+                _switchBackToMainContent(currentEvent);
```

Comment:
The duration attribute shouldn't determinate the duration of the alternative content. It's the window (PTA, Duration) to trigger the alternative content.